opentui

thdxr opened the pull request.

luisrudge commented:
```
bun x opencode-ai@opentui
```

when I press option+delete, it should delete word by word but it deletes char by char (works fine on the non-opentui version)

luisrudge commented:
I can't paste text or images into the input box

thdxr committed:
tui: fix bash command output display to prevent flickering and improve rendering performance

thdxr committed:
tui: optimize bash output processing to reduce redundant ANSI stripping operations

thdxr committed:
tmp disable bash parsing

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: enable text copying via mouse selection across all platforms including tmux/SSH

thdxr committed:
opentui: single commit from dev

thdxr committed:
core: fix snapshot revert to properly delete files when using relative paths

Fixes snapshot revert functionality so it correctly removes files that were
deleted between snapshots. Previously, revert would fail to delete files because
it was trying to delete relative paths (like 'file.txt') instead of absolute
paths ('/path/to/file.txt').

This ensures snapshot revert operations work reliably across all file types
and directory structures, maintaining consistent project state when rolling
back changes.

thdxr committed:
sync

thdxr committed:
clipboard

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: make reverted messages clickable to trigger redo

Users can now click on reverted message blocks to instantly restore
the conversation state, eliminating the need to remember keyboard
shortcuts or type /redo commands. The hover effect provides visual
feedback that the area is interactive.

thdxr committed:
tui: clarify revert description and add confirmation before redo

- Makes 'Revert' button description clearer for users ('undo messages and file changes')
- Adds confirmation dialog to prevent accidental redo of reverted messages
- Improves safety when users want to restore previously reverted work

thdxr committed:
tui: add session forking to let users branch conversations at any point in history

thdxr committed:
tui: fix dialog select layout to prevent text overflow and improve spacing

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: fix dialog text overflow and improve home screen layout for better readability

- Prevent long option text from overflowing in dialog select by truncating at 62 characters\n- Add responsive padding to home screen to prevent content touching edges\n- Change prompt container to use percentage width for better responsiveness

thdxr committed:
tui: fix session menu item states and clarify undo action

- Share/unshare menu items now properly enable/disable based on session state
- 'Undo' renamed to 'Undo previous message' for clarity
- Redo option disabled when no revert available
- Dialog filter fixed to show correct available options

thdxr committed:
tui: preserve prompt history so users can navigate previous inputs with arrow keys

thdxr committed:
tui: add timeline navigation so users can quickly jump to any previous message in long sessions

thdxr committed:
tui: fix redo menu option so users can actually redo reverted messages when session data exists

thdxr committed:
tui: rename dialog title from 'Message Actions' to 'Timeline' so users understand this is for navigating session history

thdxr committed:
tui: remove try-catch from timeline navigation so scrolling to messages works reliably without silent failures

thdxr committed:
tui: add leader+t keybind for timeline navigation so users can quickly access session history without hunting through menus

thdxr committed:
tui: prioritize opencode provider in model selection so users see their default provider first instead of hunting through alphabetically sorted list

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

actions-user committed:
chore: format code

thdxr committed:
tui: show real-time MCP server status in home screen with connection state indicators

- Displays MCP servers with visual status indicators (connected/failed/disabled)
- Shows error messages for failed connections to help users troubleshoot
- Replaces static help text with dynamic keybind display that updates based on user config
- Adds weather and puppeteer MCP servers to example configuration

thdxr committed:
add free workspace

thdxr committed:
tui: make dialog select height adapt to terminal size instead of fixed 10 lines

Previously the dialog select component had a hardcoded maxHeight of 10,
which could feel cramped on larger terminals or overwhelming on smaller
ones. Now it dynamically calculates the optimal height based on the
terminal dimensions, showing up to half the screen height minus some
padding for better visual balance.

thdxr committed:
tui: remove unused useRenderer import to clean up dialog select component

thdxr committed:
tui: hide recent models when searching to reduce visual clutter and improve model selection experience

thdxr committed:
tui: remove unused imports to fix TypeScript warnings

thdxr committed:
tui: add session sidebar with file changes, cost tracking and responsive layout

tui: show real-time MCP server status on home screen with connection indicators

tui: create dedicated logo component and improve home screen layout

tui: make session layout responsive with sidebar visible on wide terminals

thdxr committed:
tui: map home/end to jump to first/last message for faster review

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: add toggleable sidebar with 3-state control

Users can now control sidebar visibility through the command bar (Ctrl+P → 'Toggle sidebar'). The sidebar has three states:
- auto: Automatically shows/hides based on terminal width
- show: Forces sidebar visible
- hide: Forces sidebar hidden

When toggling from 'auto', it intelligently switches to the opposite of the current terminal size state.

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: remove bullet point from help menu items for cleaner display

thdxr committed:
tui: add LSP server status display to sidebar and status dialog

- Add LSP section to TUI sidebar showing connected servers with root paths
- Add LSP section to status dialog showing server count and details
- Add /lsp endpoint to server API returning server connection status
- Add lsp.updated event to notify when LSP servers are spawned
- Update SDK types to include LSP status data structures

This provides users with real-time visibility into Language Server Protocol connections alongside existing MCP status, making it easier to monitor and troubleshoot language tooling in their workspace.

thdxr committed:
tui: improve sidebar visibility logic and session title display

- Fix sidebar visibility calculation to prevent layout issues
- Add session title and share URL to sidebar header
- Remove empty text parts from assistant messages
- Adjust user message spacing for first message
- Add OpenAI codex auth plugin to configuration

thdxr committed:
tui: hide MCP section when no servers are configured and use cached data

- Only show MCP section in sidebar and status dialog when MCP servers exist
- Use cached MCP data from sync context instead of making separate API calls
- Improves performance by reducing redundant network requests

thdxr committed:
tui: improve agent tab separator visual with powerline symbol for better session distinction

thdxr committed:
test: ensure snapshot tests run reliably by preventing shared shadow repo state across parallel test execution

thdxr committed:
test: update keybind tests to match new leader key format with space separator

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
types

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
tui: improve tool rendering to show all tool parameters and fallback for unregistered tools

- Add GenericTool component to display tools without custom renderers
- Show input parameters for all tools (read, glob, grep, edit)
- Display search path and match counts for glob/grep tools
- Rename ToolRegistry.ready to render for clarity
- Add weather MCP server configuration

thdxr committed:
tui: update generic tool icon to gear for better visual distinction

thdxr committed:
tui: improve edit tool layout to adapt content display based on available terminal width

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
tui: add sidebar toggle keybind to show/hide session sidebar

Users can now press Ctrl+X followed by B to toggle the sidebar
visibility in the TUI session view, providing quick access to
session information and controls without needing to use the command
palette.

thdxr committed:
tui: fix clipboard copy functionality in tmux sessions

- Add proper tmux DCS passthrough wrapper for OSC52 sequences
- Users can now copy text to clipboard when running TUI inside tmux
- Previously clipboard copy would fail in SSH/tmux environments

thdxr committed:
tui: add command dialog integration to prompt component for external editor access

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: prevent memory leaks by properly clearing instance cache and state entries on disposal

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
thread

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
handle uncaughts

thdxr committed:
default to thread mode

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
fix

thdxr committed:
tui: respect $EDITOR environment variable and persist editor changes back to prompt

thdxr committed:
tui: extract editor opening logic into reusable utility for better code organization

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: replace manual worker message passing with structured RPC client for better type safety and reliability

thdxr committed:
tui: fix command prompt not clearing after submitting slash commands

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: improve reliability and configuration options

- Add retry logic for spawn process to handle transient failures
- Make server port and hostname configurable instead of hardcoded
- Ensure proper cleanup order in app exit handler
- Better error handling for process exit codes

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
async sync

thdxr committed:
ci: fix

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
default model

thdxr committed:
add status

thdxr committed:
exit

thdxr committed:
autoupgrade

thdxr committed:
sync

thdxr committed:
tui: clear command dialog when opening editor

Fixes issue where command dialog remains visible after opening external editor, improving user experience by ensuring clean UI transition.

thdxr committed:
tui: fix mcp server count to show singular form when only one server is connected

thdxr committed:
tui: strip ANSI escape codes from bash command output for cleaner display

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
desktop: disable fetch timeout to prevent long-running operations from being cancelled

thdxr committed:
tui: remove worker.terminate() to prevent process termination during shutdown

thdxr committed:
core: add bun fallback for install scripts to ensure compatibility across environments

thdxr committed:
tui: fix model dialog crash when provider data is missing to prevent UI errors during model selection

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
tui: disable request timeout to prevent long-running operations from being interrupted

thdxr committed:
fix

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: add event system for TUI command execution and improve SDK integration

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

geril07 committed:
[opentui] fix: dev script cannot find jsx import source (#3257)

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
tui: show queued status for user messages waiting for assistant response

thdxr committed:
tui: sort sessions alphabetically in session list for better navigation

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: fix model validation to prevent invalid provider/model combinations

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: preserve current input when navigating prompt history

thdxr committed:
sync

thdxr committed:
tui: rename option to meta in keybind system for consistency

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

rekram1-node committed:
adjust changelog generation

thdxr committed:
upgrade opentui

thdxr committed:
tui: fix text wrapping to prevent layout overflow in terminal UI

thdxr committed:
sidebar

berenar committed:
refactor: whitelist some safe bash tools in Plan agent (#3288)

Co-authored-by: Aiden Cline <aidenpcline@gmail.com>

veracioux committed:
fix: Opencode fails with ENOENT posix_spawn '/usr/bin/rg' (#3396)

thdxr committed:
tui: add 21-language syntax highlighting support with tree-sitter parsers

Enable syntax highlighting for Python, TypeScript, JavaScript, Rust, Go, C++, C#, Java, Ruby, PHP, Scala, HTML, CSS, Haskell, Julia, OCaml, Bash, C, JSON, JSX, and TSX. Users can now view properly highlighted code across all popular programming languages instead of plain text.

thdxr committed:
core: expand syntax highlighting from 1 to 21 languages for better code readability

Updated parser configuration to support highlighting for Python, TypeScript, JavaScript, Rust, Go, C++, C#, Java, Ruby, PHP, Scala, HTML, CSS, Haskell, Julia, OCaml, Bash, C, JSON, JSX, and TSX. Users viewing code in the editor now get proper syntax highlighting instead of monochrome text across all popular languages.

thdxr committed:
tui: improve file path display in modified files list with proper truncation and line wrapping

Updated the session sidebar to intelligently truncate long file paths so they fit properly in the UI without breaking layout. File names stay visible while directory paths are shortened, and text now wraps character-by-character when needed to prevent overflow in narrow terminals.

thdxr committed:
update

thdxr committed:
Opentui textarea (#3310)

Co-authored-by: Sebastian Herrlinger <hasta84@gmail.com>

thdxr committed:
Revert "Opentui textarea" (#3398)

thdxr committed:
update codex prompt

Commit:
release: v0.15.15

thdxr committed:
codex does not have reasoning effort set

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
fix: prevent crash when pressing up arrow with no history

thdxr committed:
fix: copy share URL to clipboard when using /share command in TUI

thdxr committed:
tui: fix text wrapping in share URL and context display to prevent overflow

zendril committed:
fix: windows not loading due to getContextId non-hydrating error (#3453)

thdxr committed:
Opentui textarea (#3399)

Co-authored-by: Sebastian Herrlinger <hasta84@gmail.com>
Co-authored-by: GitHub Action <action@github.com>

kommander committed:
load actual opentui version for native prebuild packages

kommander committed:
placeholder

actions-user committed:
chore: format code

kommander committed:
show textarea placeholder only on home

veracioux committed:
opentui: add toast & port CLI flags (-c, -s, -m --agent) (#3447)

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
sync

rekram1-node committed:
Merge branch 'dev' into opentui

rekram1-node committed:
fix: port mcp logic

kommander committed:
linefeed is newline by default

kommander committed:
use keybinds

kommander committed:
new opentui snapshot

actions-user committed:
chore: format code

kommander committed:
expand pasted content into prompt

kommander committed:
threshold for collapsing pasted text

kommander committed:
instert custom commands

kommander committed:
use new snapshot

actions-user committed:
chore: format code

kommander committed:
up/down history

thdxr committed:
tui: add image paste support from clipboard

Users can now paste images directly from their clipboard into the prompt using the paste command. The image will be automatically converted to base64 and embedded as a file attachment. This works on macOS by reading PNG data from the clipboard and converting it to the appropriate format for the AI to process.

thdxr committed:
tui: allow pasting image file paths to automatically attach images to prompt

thdxr committed:
sync

kommander committed:
use opentui tabstop snapshot

actions-user committed:
chore: format code

kommander committed:
enable @filename parts within text, not just at end

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: remove deprecated and unused keybinds to simplify configuration

Users no longer see confusing deprecated keybind options in config that were never implemented (switch_mode, file operations, etc). Keybinds now use consistent Zod schema defaults instead of hardcoded fallbacks, making configuration more predictable. Session interruption (esc) now works consistently through keybind system.

thdxr committed:
sync

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
Merge branch 'dev' into opentui

kommander committed:
allow @ completion at beginning of newline

kommander committed:
use latest opentui snapshot

thdxr committed:
fix interrupt

rgodha24 committed:
feat(opentui): port theming  (#3234)

Co-authored-by: Dax <mail@thdxr.com>
Co-authored-by: Dax Raad <d@ironbay.co>

thdxr committed:
fix extmark highlight

thdxr committed:
fix autocomplete flashing by maintaining previous results during refetch

thdxr committed:
Merge branch 'dev' into opentui

thdxr committed:
tui: ensure autocomplete borders match selected theme

kommander committed:
use opentui unicode snapshot

rekram1-node committed:
fix

veracioux committed:
opentui: fix: Show toasts while in session (#3591)

veracioux committed:
opentui: fix: Dialog button clicks leak through (#3589)

veracioux committed:
opentui: fix: Dialog command evt.preventDefault() (#3587)

thdxr committed:
Merge branch 'dev' into opentui

rekram1-node committed:
fix

thdxr committed:
Merge branch 'dev' into opentui

Commit:
release: v0.15.30

thdxr committed:
sync

thdxr committed:
tui: fix help dialog trigger to use command system instead of direct dialog

thdxr committed:
tui: improve text formatting in summary prompt to ensure proper XML wrapping

thdxr committed:
tui: fix summary prompt text formatting to ensure proper XML structure

thdxr committed:
tui: add debug logging for summary body to help troubleshoot response formatting

adamdotdevin committed:
wip: desktop work

actions-user committed:
ignore: update download stats 2025-10-31

adamdotdevin committed:
wip: desktop work

adamdotdevin committed:
wip: desktop work

adamdotdevin committed:
wip: desktop work

veracioux committed:
opentui: fix: Implement session delete endpoint (#3601)

veracioux committed:
opentui: feat: Implement POST /tui/toast-show (#3604)

rekram1-node committed:
tweak: filter out deprecated models

rekram1-node committed:
update types

iamdavidhill committed:
Added border as shadow and updated button

Commit:
zen: billing page layout

S-A-Martin committed:
tweak - normalise unix-like identifiers to support git bash for windows (#2100)

Co-authored-by: Steven Martin <smartin@clearcom.com>

adamdotdevin committed:
wip: desktop work

adamdotdevin committed:
wip: desktop work

iamdavidhill committed:
wip: Tweaking button styles

iamdavidhill committed:
wip: Tweaking select styles

This ghost style is placeholder for now until we style this properly with an icon and all the states.

thdxr committed:
fix tests

kommander committed:
setup markdown rendering

thdxr committed:
fix tests

thdxr committed:
cleanup

thdxr committed:
tui: add text selection and copy functionality to messages

thdxr committed:
Merge branch 'dev' into opentui

veracioux committed:
fix: Missing @opencode-ai/plugin causes crash (#3615)

Commit:
sync

veracioux committed:
opentui: fix: Deleting current session from TUI (#3617)

Commit:
release: v0.15.31

cogni-ai-ee commented on logs/mcp-puppeteer-2025-10-07.log:
I think these logs shouldn't be committed (the whole logs/).
